src/components/SelectionList/SearchTableHeader.tsx

```diff
@@ -103,52 +103,50 @@ type SearchTableHeaderProps = {
 };
 
 function SearchTableHeader({data, metadata, sortBy, sortOrder, onSortPress, shouldShowYear, shouldShowSorting}: SearchTableHeaderProps) {
-    const styles = useThemeStyles();
-    const StyleUtils = useStyleUtils();
     // eslint-disable-next-line rulesdir/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth
     const {isSmallScreenWidth, isMediumScreenWidth} = useResponsiveLayout();
-    const {translate} = useLocalize();
     const displayNarrowVersion = isMediumScreenWidth || isSmallScreenWidth;
 
-    if (SearchColumns[metadata.type] === null) {
-        return;
-    }
+    const shouldShowColumn = useCallback(
+        (columnName: SearchColumnType | typeof CONST.REPORT.TRANSACTION_LIST.COLUMNS.COMMENTS) => {
+            if (columnName === CONST.REPORT.TRANSACTION_LIST.COLUMNS.COMMENTS) {
+                return false;
+            }
```

Comment:
Can we add this to the `shouldShowColumnConfig` so we don't split the logic?

Reply:
Yes we can do that 👍 also provided more extensive answer in the other comment below